# This is a BitKeeper generated diff -Nru style patch.
#
# ChangeSet
#   2005/02/06 21:47:40-08:00 shemminger@osdl.org 
#   [PKT_SCHED]: netem: memory leak
#   
#   Good catch.. netem needs to free skb's that are dropped due to loss
#   simulation.
#   
#   Signed-off-by: David S. Miller <davem@davemloft.net>
# 
# net/sched/sch_netem.c
#   2005/02/06 21:47:26-08:00 shemminger@osdl.org +1 -0
#   [PKT_SCHED]: netem: memory leak
# 
diff -Nru a/net/sched/sch_netem.c b/net/sched/sch_netem.c
--- a/net/sched/sch_netem.c	2005-02-09 09:13:00 -08:00
+++ b/net/sched/sch_netem.c	2005-02-09 09:13:00 -08:00
@@ -180,6 +180,7 @@
 	if (q->loss && q->loss >= get_crandom(&q->loss_cor)) {
 		pr_debug("netem_enqueue: random loss\n");
 		sch->stats.drops++;
+		kfree_skb(skb);
 		return 0;	/* lie about loss so TCP doesn't know */
 	}
 
